src/main/java/org/springframework/integration/aws/outbound/SnsMessageHandler.java

```diff
@@ -113,6 +120,21 @@ public void setSubjectExpression(Expression subjectExpression) {
 		this.subjectExpression = subjectExpression;
 	}
 
+	public void setMessageGroupId(String messageGroupId) {
+	    Assert.hasText(messageGroupId, "messageGroupId must not be empty.");
+	    this.messageGroupIdExpression = new LiteralExpression(messageGroupId);
+	}
+
+	public void setMessageGroupIdExpression(Expression messageGroupIdExpression) {
+	    Assert.notNull(messageGroupIdExpression, "messageGroupIdExpression must not be null.");
+	    this.messageGroupIdExpression = messageGroupIdExpression;
+	}
+
+	public void setMessageDeduplicationIdExpression(Expression messageDeduplicationIdExpression) {
+	    Assert.notNull(messageDeduplicationIdExpression, "messageDeduplicationIdExpression must not be null.");
+		this.messageDeduplicationIdExpression = messageDeduplicationIdExpression;
```

Comment:
Something is wrong with formatting.
Not sure what is the reason in the extra indent on this line.